File: SLMS_UI/src/app/core/models/CreateLibraryRequest .ts
import { InstitutionStatus } from "@core/enums/OnbardingSteps";

export interface CreateLibraryRequest {
  name: string;
  description: string;
  address: string;
  email?: string;
  phone?: string;
  floor: number;
  capacity: number;
  isActive: boolean;
  isPrimary: boolean;
  isOnboarding: boolean;
  status: InstitutionStatus
}

File: SLMS_API/Application/Contracts/Organizations/Requests/CreateLibraryRequest.cs
using SLMS_API.Common.Enums;

namespace SLMS_API.Application.Contracts.Organizations.Requests;

public class CreateLibraryRequest
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Address { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public int? Floor { get; set; }
    public int? Capacity { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsPrimary { get; set; } = true;
    public bool IsOnboarding { get; set; } = false;
    public InstitutionStatus Status { get; set; } = InstitutionStatus.Active;
}
